samples/webforms/embed-a-dashboard-into-an-app/PBIWebApp/Default.aspx.cs

```diff
@@ -0,0 +1,189 @@
+            request.Method = "GET";
+            request.ContentLength = 0;
+            request.Headers.Add("Authorization", String.Format("Bearer {0}", authResult.AccessToken));
+
+            //Get dashboards response from request.GetResponse()
+            using (var response = request.GetResponse() as System.Net.HttpWebResponse)
+            {
+                //Get reader from response stream
+                using (var reader = new System.IO.StreamReader(response.GetResponseStream()))
+                {
+                    responseContent = reader.ReadToEnd();
+
+                    //Deserialize JSON string
+                    PBIDashboards PBIDashboards = JsonConvert.DeserializeObject<PBIDashboards>(responseContent);
+                    Table1.Visible = true;
+                    //Get each dashboard 
+                    foreach (PBIDashboard dsb in PBIDashboards.value)
+                    {
+                        TableRow tRow = new TableRow();
+
+                        Table1.Rows.Add(tRow);
+                        TableCell idCell = new TableCell();
+                        idCell.Text = dsb.id;
+                        tRow.Cells.Add(idCell);
+
+                        Table1.Rows.Add(tRow);
+                        TableCell nameCell = new TableCell();
+                        nameCell.Text = dsb.displayName;
+                        tRow.Cells.Add(nameCell);
+
+                        Table1.Rows.Add(tRow);
+                        TableCell isReadOnlyCell = new TableCell();
+                        isReadOnlyCell.Text = dsb.isReadOnly.ToString();
+                        tRow.Cells.Add(isReadOnlyCell);
+
+                        Table1.Rows.Add(tRow);
+                        TableCell embedUrlCell = new TableCell();
+                        embedUrlCell.Text = dsb.embedUrl;
+                        tRow.Cells.Add(embedUrlCell);
+                    }
+                }
+            }
+        }
+
+
+    }
+    
+    //Power BI Datasets
+    public class PBIDatasets
+    {
+        public Dataset[] Datasets { get; set; }
+    }
+
+    public class PBIGroups
+    {
+        public PBIGroup[] value { get; set; }
+    }
+
+    public class Dataset
+    {
+        public string Id { get; set; }
+        public string Name { get; set; }
+    }
+
+    public class PBIGroup
+    {
+        public string id { get; set; }
+        public string name {get; set;}
+    }
+
+    public class PBIDashboards
+    {
+        public PBIDashboard[] value { get; set; }
+    }
+    public class PBIDashboard
+    {
+        public string id { get; set; }
+        public string displayName { get; set; }
+        public string embedUrl { get; set; }
+        public bool isReadOnly { get; set; }
+    }
+    public class PBIReports
+    {
+        public PBIReport[] value { get; set; }
+    }
+    public class PBIReport
```

Comment:
resolved (and I deleted all the other defenitions that are not used in this sample as well)